app/assets/images/homepage/hero_images/2x/resize_to_2000.sh

```diff
@@ -0,0 +1,9 @@
+max_size=2000
+width=$(vipsheader -f width $1)
+height=$(vipsheader -f height $1)
+size=$((width > height ? width : height))
+factor=$(bc <<< "scale=10; $max_size / $size")
+
+new_name=$(sed 's/\.jpg/_new\.jpg/' <<< "$1")
+
+vips resize $1 $new_name $factor
```

Comment:
I don't understand the "scale=10" and all this other arithmetic...

Can't you just tell `vips resize` to resize the original to 2000 pixels wide, and let it do the arithmetic?

Ah, the `resize` command annoyingly won't, but how about just use `vipsthumbnail` instead?  This also lets you convert from an original image format (say tiff) to jpg in one go.... while resizing from a full-size JPG as source is almost certainly fine in my experience, technically it's lossy-to-lossy and not best practice. 

    vipsthumbnail input.tiff --size 2000x65500 -o output.jpg

(just having the "height" be a huge number to tell it to resize width to 5000 while keeping aspect ratio is what works for me, and was necessary in older versions of vips, although might be possible to leave off the height now)